main.cpp

```diff
@@ -19,13 +19,26 @@ using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
 public:
-    CastCallBack(Rewriter& rewriter) {
-        // Your code goes here
-    };
+    CastCallBack(Rewriter& rewriter) : _rewriter(rewriter) {};
+    void run(const MatchFinder::MatchResult& Result) override {
+        const auto* cast_expression = Result.Nodes.getNodeAs<CStyleCastexpr>("cast");
+        auto replace = CharSourceRange::getCharRange(cast_expression->getLParenLoc(),
+            cast_expression->getSubexprAsWritten()->getBeginLoc());
+        auto& src_mngmnt = *Result.SourceManager;
+        auto typename = Lexer::getSourceText(CharSourceRange::getTokenRange(

```

Comment:
Please descrbe the difference between the use of `getCharRange()` and `getTokenRange()` with examples.

Reply:
- "`getCharRange()` returns range specifying the starting/ending character by the specified locations."
- "`getTokenRange()` returns range specifying the start of the range and the start of the last token of the range."